### PR TITLE
🐛 Fixed correct filtering of `page` when `fields` query param is used

### LINKED ISSUE #10351: Content API returns `page` when `fields` query param is used
When making a request to the `/posts/:id` or `/pages/:id` endpoint and you request specific fields, e.g. `title` you will also get back the `page` boolean.

### To replicate
  - start ghost
  - create an integration
  - copy content key
  - curl http://localhost:2368/ghost/api/v2/posts/?key=$key | jq .posts[0].id
  - copy the id
  - curl "http://localhost:2368/ghost/api/v2/posts/$id/?key=$key&fields=title"  | jq .

### Desired behaviour

The `page` param should not be sent back!

### PR DESCRIPTION
**Pull request by rshbhgrg**
closes #10351

- Updated filtering logic for `page` in input serialisers of post and page in v2

### CODE REVIEW

File: core/server/api/v2/utils/serializers/input/pages.js [hunk 1]
```diff
@@ -34,7 +34,19 @@ module.exports = {
     read(apiConfig, frame) {
         debug('read');
 
-        frame.data.page = true;
+        /**
```
[the remark this answers is not in the document]
Comment (rishabhgrg) [reply core/server/api/v2/utils/serializers/input/pages.js]: @gargol Yeah makes sense, have moved over to a common method now 😄

**Comment by gargol** [comment]
The tests seem to have broken? Did you have a chance looking into it @rishabhgrg? :)

**Comment by rishabhgrg** [comment]
@gargol Updated PR with discussed solution as the original solution hit a complete roadblock, have updated details [here](https://github.com/TryGhost/Ghost/issues/10351#issuecomment-453475591). Tests are all ✅ now

File: core/server/api/v2/utils/serializers/input/pages.js [hunk 2]
```diff
@@ -8,24 +8,27 @@ function removeMobiledocFormat(frame) {
...
-         *
-         * @TODO: https://github.com/TryGhost/Ghost/issues/10268
-         */
-        if (frame.options.filter) {
-            frame.options.filter = `${frame.options.filter}+page:true`;
-        } else {
-            frame.options.filter = 'page:true';
-        }
-
+        addPageFilter(frame);
```
Comment (gargol) [inline core/server/api/v2/utils/serializers/input/pages.js]: Now that this method is only used once. Let's just keep it as was, so the PR is just focused on one specific change? ;)

**Comment by gargol** [comment]
Will have a closer look at this now :+1:

**Comment by gargol** [comment]
The column cleanup logic now will effect both Content and Admin API, is that an effect we are looking for? If we are only to filter this out in Content API putting the cleanup condition into ` if (utils.isContentAPI(frame)) {` above it would be betters suited :thinking:

**Comment by rishabhgrg** [comment]
> The column cleanup logic now will effect both Content and Admin API, is that an effect we are looking for?

I thought about this case before putting it outside the `isContentAPI` condition, and I feel this cleanup should be for both APIs as this is more of a patch covering for a bookshelf issue and not a content api specific feature/change. Ideally, if an API users explicitly passes `fields=title` in the query, they'll expect only `title` to be returned (and well `id` i guess). In that way it makes sense for `fields` on query to work consistently across both APIs, wdyt?

**Review by naz — APPROVED** [review]

**Pull request merged**